src/native/windows/apps/prunsrv/prunsrv.c

```diff
@@ -1380,7 +1388,10 @@ static DWORD WINAPI serviceStop(LPVOID lpParameter)
 
     if (dwCtrlType == SERVICE_CONTROL_SHUTDOWN)
         timeout = MIN(timeout, apxGetMaxServiceTimeout(gPool));
-    reportServiceStatus(SERVICE_STOP_PENDING, NO_ERROR, timeout);
+    if (timeout)
+        reportServiceStatus(SERVICE_STOP_PENDING, NO_ERROR, timeout);
+    else
+        reportServiceStatus(SERVICE_STOP_PENDING, NO_ERROR, ONE_MINUTE_SEC);
 
     if (timeout) {
         FILETIME fts, fte;
```

Comment:
Again, isn't timeout now always >0 ?

Reply:
The corresponding  logic has been rewritten.